lib/pos_mempool.go

```diff
@@ -527,32 +541,27 @@ func (mp *PosMempool) validateTransaction(txn *MsgDeSoTxn, verifySignature bool)
 		return errors.Wrapf(err, "PosMempool.AddTransaction: Problem validating transaction nonce")
 	}
 
-	if !verifySignature {
-		return nil
+	return nil
+}
+
+func (mp *PosMempool) updateTransactionValidatedStatus(txnHash *BlockHash, validated bool) {
+	mp.Lock()
+	defer mp.Unlock()
+
+	if !mp.IsRunning() || txnHash == nil {
+		return
 	}
 
-	// Check transaction signature.
-	if _, err := mp.readOnlyLatestBlockView.VerifySignature(txn, uint32(mp.latestBlockHeight)); err != nil {
-		return errors.Wrapf(err, "PosMempool.AddTransaction: Signature validation failed")
+	txn := mp.txnRegister.GetTransaction(txnHash)
+	if txn == nil {
+		return
 	}
 
-	return nil
+	txn.SetValidated(validated)
```

Comment:
Since this is a write-lock, I'm worried this could cause a bottleneck. Just to max efficiency, I would try and do something more like this if you can:
```
// 1. With no lock, or just with RLock, get all the txns you want to mark as validated.
txnsToMarkAsValidated = // whatever
mp.Lock()
// 2. Inside the lock blitz through these txns and call txn.SetValidated() on all of them in one go
mp.Unlock()
```

Reply:
It almost feels like you shouldn't even need to lock the whole mempool to update a single txn like this but if you're going to do it I'd just try to minimize the time spent holding the lock a little bit more, as I mentioned above.